engine.py

```diff
@@ -357,10 +359,26 @@ def execute(cls, ids, data):
             address_with_party=False):
             if model:
                 records = cls._get_dual_records(ids, model, data)
+
+            # report single and len > 1, return zip file
+            if action.single and len(ids) > 1:
+                content = BytesIO()
+                with zipfile.ZipFile(content, 'w') as content_zip:
+                    for record in records:
+                        oext, rcontent = cls._execute_html_report([record],
+                            data, action)
+                        rfilename = '%s-%s.%s' % (
+                            action_name,
+                            (record.render.rec_name).replace('/', '-'),
```

Comment:
Should we use a more sophisticated method to ensure the filename is valid?

IE, I found this:

https://gist.github.com/dbr/256270